packages/debugger-extension/src/index.ts

```diff
@@ -719,24 +720,37 @@ const main: JupyterFrontEndPlugin<void> = {
     });
 
     commands.addCommand(CommandIDs.pauseOnExceptions, {
-      label: trans.__('Enable / Disable pausing on exceptions'),
-      caption: () =>
-        service.isStarted
-          ? service.pauseOnExceptionsIsValid()
-            ? service.isPausingOnExceptions
-              ? trans.__('Disable pausing on exceptions')
-              : trans.__('Enable pausing on exceptions')
-            : trans.__('Kernel does not support pausing on exceptions.')
-          : trans.__('Enable / Disable pausing on exceptions'),
-      className: 'jp-PauseOnExceptions',
-      icon: Debugger.Icons.pauseOnExceptionsIcon,
-      isToggled: () => {
-        return service.isPausingOnExceptions;
-      },
+      label: args => (args.filter as string) || 'Breakpoints on exception',
+      caption: args => args.description as string,
+      isToggled: args =>
+        service.session?.isPausingOnException(args.filter as string) || false,
       isEnabled: () => service.pauseOnExceptionsIsValid(),
-      execute: async () => {
-        await service.pauseOnExceptions(!service.isPausingOnExceptions);
-        commands.notifyCommandChanged();
+      execute: async args => {
+        let filter: string | undefined;
+        if (args?.filter) {
+          filter = args.filter as string;
+        } else {
+          let items: string[] = [];
+          items.push('None');
+          service.session?.exceptionBreakpointFilters?.forEach(
+            availableFilter => {
+              items.push(availableFilter.filter);
+            }
+          );
+          filter =
+            (
+              await InputDialog.getItem({
+                title: trans.__('Select a filter for breakpoints on exception'),
+                items: items,
+                current: service.session?.currentExceptionFilter || 'None',
+                editable: false
+              })
+            ).value ?? undefined;
+        }
+        if (!filter) {
+          return;
+        }
+        await service.pauseOnExceptions(filter as string);
```

Comment:
The input dialog return the same object as a `Dialog`. So you should prefer checking if the button pressed was accept:

```suggestion
          const result = await InputDialog.getItem({
                title: trans.__('Select a filter for breakpoints on exception'),
                items: items,
                current: service.session?.currentExceptionFilter || 'None',
                editable: false
              });
          filter = result.button.accept ? result.value : null;
        }
        if (!filter) {
          return;
        }
        await service.pauseOnExceptions(filter as string);
```